sections/Animation/Animation.tsx

```diff
@@ -0,0 +1,154 @@
+import { Section } from "deco/blocks/section.ts";
+import { useId } from "$store/sdk/useId.ts";
+
+interface Children {
+  label?: string;
+  section: Section;
+}
+
+interface Props {
+  animationType?:
+    | "fade-in"
+    | "fade-in-bottom"
+    | "slide-left"
+    | "slide-right"
+    | "zoom-in";
+  /**
+   * @default 0.3
+   */
+  duration?: string;
+  children: Children;
+}
+
+function Animation(
+  { children, animationType = "fade-in", duration = "0.3" }: Props,
+) {
+  const { section } = children;
+
+  const { Component, props } = section;
+  const id = useId();
+
+  const animationClasses = {
+    "fade-in": "animate-fade-in",
+    "fade-in-bottom": "animate-fade-in-bottom",
+    "slide-left": "animate-slide-left",
+    "slide-right": "animate-slide-right",
+    "zoom-in": "animate-zoom-in",
+  };
+
+  const animationClass = animationClasses[animationType];
+
+  return (
+    <>
+      <style
+        dangerouslySetInnerHTML={{ __html: animationByType[animationType] }}
+      >
+      </style>
+      <div
+        id={id}
+        class="opacity-0"
+        style={{ animationDuration: duration + "s" }}
+      >
+        <Component {...props} />
+      </div>
+      <script
+        dangerouslySetInnerHTML={{
+          __html: `
+                var observer = new IntersectionObserver(function(entries) {
+                    entries.forEach(entry => {
+                        if (entry.isIntersecting) {
+                            entry.target.classList.add("${animationClass}");
+                            entry.target.classList.remove("opacity-0");
+                            observer.disconnect();
+                        }
+                    });
+                }, { threshold: 0.50 });
+            
+                var element = document.getElementById('${id}');
+                observer.observe(element);
+            `,
+        }}
+      />
```

Comment:
This will make the page break the rendering due to script. What do you think about adding this into the `<head>`?

Reply:
I inserted async now

Reply:
async doesn't change for inline script